### ISSUE TITLE
[Held requests] When partial approving report, long delay creating new report for held expense(s)

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#41652]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.70-2
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @JmillsExpensify 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1714761240737509

## Action Performed:

1. Navigate to a report containing several expenses from someone else
2. Click in the three dot menu in the expense
3. Choose Hold, leave comment
4. Confirm that the expense is on hold (e.g. Hold in the header)
5. Navigate to the report, click Approve
6. Choose Approve only…

## Expected Result:
A new report with the held expense is created immediately in the front-end
## Actual Result:
A new report with the held expense is created 10+ seconds later
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [X] Android: Native
- [X] Android: mWeb Chrome
- [X] iOS: Native
- [X] iOS: mWeb Safari
- [X] MacOS: Chrome / Safari
- [X] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/38435837/2da79e50-ce70-4195-affb-37d1ed617a04

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @kevinksullivan</details>

**Comment by JmillsExpensify** [comment]
Screenshots for clarity

**Report created optimistically with multiple currencies**
![CleanShot 2024-05-06 at 06 32 31@2x](https://github.com/Expensify/App/assets/1127863/9326b325-11b1-4b3b-bade-397df474e856)

**Report preview appropriately showing 2 expenses**
![CleanShot 2024-05-06 at 06 31 47@2x](https://github.com/Expensify/App/assets/1127863/d31695c3-9520-4683-9d8b-544519999bdb)

**Comment by JmillsExpensify** [comment]
Based on a Slack convo, I think that @war-in might be interested in this issue?

**Comment by nkdengineer** [comment]
@JmillsExpensify This bug is the same as https://github.com/Expensify/App/issues/39338. I think we should re-open the original issue.

**Comment by war-in** [comment]
> Based on a Slack convo, I think that @war-in might be interested in this issue?

Yes, I would like to take this one

**Comment by trjExpensify** [comment]
@war-in are you good to prioritise putting up a PR for this one?

**Comment by trjExpensify** [comment]
@nkdengineer @robertjchen I'm going to assign you both to ultimately review the PR. 👍

**Comment by war-in** [comment]
Yes, however, I work part-time and will be available on Friday. Is that okay with you?

**Comment by robertjchen** [comment]
In progress/Discussions on Slack

**Comment by war-in** [comment]
Hi @robertjchen 👋 
Contacting you there because I was removed from the `#expensify-bugs` channel 😢 
My current progress looks like that

https://github.com/Expensify/App/assets/61014013/66621ff5-91dc-4535-ae86-cea62ac7bfb3

It's not finished yet but would be nice to get an endpoint which accepts the `reportID` parameter and does not create a new one. Do you have an ETA for it? Or maybe such an endpoint already exists?

**Comment by robertjchen** [comment]
Finishing automated tests for backend, should be open for review by EOD

**Comment by robertjchen** [comment]
**Partial Pay/Approve Optimistic Flow ([GH](https://github.com/Expensify/App/issues/41652))**

- **Partial Approve**
    - ~Auth [PR](https://github.com/Expensify/Auth/pull/10961) deployed~
    - Web [PR](https://github.com/Expensify/Web-Expensify/pull/42127) on `staging`
    - Auth [PR 2](https://github.com/Expensify/Auth/pull/11028) under testing (new! coming from Slack [discussion](https://expensify.slack.com/archives/C049HHMV9SM/p1716804540535099?thread_ts=1714761240.737509&cid=C049HHMV9SM))
    - App Draft [PR](https://github.com/Expensify/App/pull/42573) in progress!

- **Partial Pay**
    - Auth [PR](https://github.com/Expensify/Auth/pull/11015) [to be deployed](https://expensify.slack.com/archives/C0714QF3A1Z/p1716808072093939)
    - Web [PR](https://github.com/Expensify/Web-Expensify/pull/42141) in draft/testing

**Comment by robertjchen** [comment]
**Partial Pay/Approve Optimistic Flow ([GH](https://github.com/Expensify/App/issues/41652))**

- **Partial Approve**
    - ~Web [PR](https://github.com/Expensify/Web-Expensify/pull/42127) deployed~
    - Auth [PR 2](https://github.com/Expensify/Auth/pull/11028) under review!

- **Partial Pay**
    - ~Auth [PR](https://github.com/Expensify/Auth/pull/11015) deployed!~
    - Web [PR](https://github.com/Expensify/Web-Expensify/pull/42141) in draft/testing to be opened for review early next week.

App Draft [PR](https://github.com/Expensify/App/pull/42573) in progress, awaiting deployment of the above. We will test out partial approve early next week, followed by partial pay.

**Misc ([SO thread](https://expensify.slack.com/archives/C036QM0SLJK/p1717127275429169?thread_ts=1716981703.521049&cid=C036QM0SLJK)):**
- [Add Hold to the comment action menu](https://github.com/Expensify/App/issues/41440) in discussion/review. May require Auth changes (onyx updates), investigating.
- [Unholding an expense isn’t optimistically reflected on the expense preview](https://expensify.slack.com/archives/C049HHMV9SM/p1717127248876809) - May require Auth changes (onyx updates), investigating.
- [Partial approved reports don’t have GBR in the LHN](https://expensify.slack.com/archives/C049HHMV9SM/p1717128748434439) - May require Auth changes (onyx updates), investigating.

**Comment by war-in** [comment]
@robertjchen I've just checked `Partial Approve` flow and it works as expected 🎉 
Now waiting for the second PR fixing double previews :)

**Comment by war-in** [comment]
I've checked `Partial Pay` and it looks like there was no update of the API 🤔 
Do you know if it was updated?

**Comment by robertjchen** [comment]
Perfect, thanks for confirming!! I was focused on getting the `Partial Approve` fix out last week, I'll work on getting the `Partial Pay` flow working this week 👍

**Comment by robertjchen** [comment]
Still working on this, aiming to get the PRs open by EOD

**Comment by robertjchen** [comment]
Ongoing discussion in https://expensify.slack.com/archives/C049HHMV9SM/p1718024770446099?thread_ts=1714761240.737509&cid=C049HHMV9SM

Finalizing backend PR for `Partial Approve` in the interim.

**Comment by robertjchen** [comment]
Working on backend PRs to address what was discussed on Slack:
- https://github.com/Expensify/Auth/pull/11229
- https://github.com/Expensify/Auth/pull/11223

**Comment by robertjchen** [comment]
Got the Partial Pay PRs merged at the end of last week. Should be out on `staging` later today Monday for testing.

**Remaining Items:**
- `ApproveReport`: backend changes for optimistic `reportActionID`'s: https://github.com/Expensify/Auth/pull/11223
- Kicking off the medium-sized backend changes in order to support the optimistic/offline behavior as agreed on in the [Slack discussion](https://expensify.slack.com/archives/C049HHMV9SM/p1718036736876909?thread_ts=1714761240.737509&cid=C049HHMV9SM).

**Comment by robertjchen** [comment]
**Remaining Items:**
- `ApproveReport`: backend changes for optimistic `reportActionID`'s: https://github.com/Expensify/Web-Expensify/pull/42384
- Started the backend changes in order to support the optimistic/offline behavior as agreed on in the [Slack discussion](https://expensify.slack.com/archives/C049HHMV9SM/p1718036736876909?thread_ts=1714761240.737509&cid=C049HHMV9SM). https://github.com/Expensify/Auth/pull/11272

**Comment by robertjchen** [comment]
**Remaining Items:**
- Tested on `staging`, working on followup fixes for `ApproveReport` double-previews issue.
- Started the backend changes in order to support the optimistic/offline behavior as agreed on in the [Slack discussion](https://expensify.slack.com/archives/C049HHMV9SM/p1718036736876909?thread_ts=1714761240.737509&cid=C049HHMV9SM). https://github.com/Expensify/Auth/pull/11272

**Comment by luacmartins** [comment]
@robertjchen I can help review the draft PR if needed since I'm familiar with the PayMoneyRequest flow.

**Comment by robertjchen** [comment]
Thanks, will keep you posted!

**Comment by robertjchen** [comment]
**Remaining Items:**
- Tested on `staging`, working on followup fixes for `ApproveReport` double-previews issue, local tests ongoing.
- backend changes in order to support the optimistic/offline behavior as agreed on in the [Slack discussion](https://expensify.slack.com/archives/C049HHMV9SM/p1718036736876909?thread_ts=1714761240.737509&cid=C049HHMV9SM). https://github.com/Expensify/Auth/pull/11272

**Comment by robertjchen** [comment]
**Remaining Items:**
- working on followup fixes for ApproveReport double-previews issue, local tests ongoing.
- backend changes in order to support the optimistic/offline behavior as agreed on in the [Slack discussion](https://expensify.slack.com/archives/C049HHMV9SM/p1718036736876909?thread_ts=1714761240.737509&cid=C049HHMV9SM). https://github.com/Expensify/Auth/pull/11272

**Comment by robertjchen** [comment]
**Remaining Items:**
- working on followup fixes for ApproveReport double-previews issue, local tests ongoing.
- backend changes in order to support the optimistic/offline behavior as agreed on in the [Slack discussion](https://expensify.slack.com/archives/C049HHMV9SM/p1718036736876909?thread_ts=1714761240.737509&cid=C049HHMV9SM). https://github.com/Expensify/Auth/pull/11272

**Comment by war-in** [comment]
Update:

I tested double-previews issue and it's gone somehow 🎉  Now waiting for optimistic report actions and once it's done my PR is ready for review

**Comment by robertjchen** [comment]
Working on backend changes in order to support the optimistic/offline behavior as agreed on in the [Slack discussion](https://expensify.slack.com/archives/C049HHMV9SM/p1718036736876909?thread_ts=1714761240.737509&cid=C049HHMV9SM). Aiming to open PR for review by EOW.

**Comment by robertjchen** [comment]
Working on backend changes in order to support the optimistic/offline behavior as agreed on in the [Slack discussion](https://expensify.slack.com/archives/C049HHMV9SM/p1718036736876909?thread_ts=1714761240.737509&cid=C049HHMV9SM). Aiming to open PR for review by Wed.

**Comment by robertjchen** [comment]
PR being tested at the moment, aiming to complete over weekend.

**Comment by robertjchen** [comment]
Implementation is complete! Working on automated tests and getting it reviewed tomorrow for deploy by Wed. 🤞

**Comment by robertjchen** [comment]
Backend Auth PR now under review!

**Comment by war-in** [comment]
Frontend draft PR is going through an internal (SWM) review. Will mark it as `ready for review` ASAP 🚀

**Comment by war-in** [comment]
PR marked as ready for review

**Comment by robertjchen** [comment]
PR was merged and on the way out!

**Comment by trjExpensify** [comment]
Assigned!

**Comment by robertjchen** [comment]
^ The above is not a blocker. [Explanation](https://github.com/Expensify/App/issues/46310#issuecomment-2255435328)
